api/server/router/network/filter.go

```diff
@@ -29,6 +29,29 @@ func filterNetworkByType(nws []types.NetworkResource, netType string) ([]types.N
 	return retNws, nil
 }
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
+	case true:
+		for _, nw := range nws {
+			if len(nw.Containers) == 0 && !runconfig.IsPreDefinedNetwork(nw.Name) {
```

Comment:
@AkihiroSuda: I'm not sure I understand correctly, but wouldn't be more straightforward to add `network` selectors to `ListTasksRequest` and `ListServicesRequest`? "List the tasks and services that depend on this network" feels better from an API perspective than "tell me if it's possible to delete this network".